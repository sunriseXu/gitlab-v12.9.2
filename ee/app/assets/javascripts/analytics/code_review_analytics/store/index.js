import Vue from 'vue';
import Vuex from 'vuex';
import * as actions from './actions';
import mutations from './mutations';
import state from './state';

Vue.use(Vuex);

const createStore = () =>
  new Vuex.Store({
    state: state(),
    actions,
    mutations,
  });

export default createStore();
