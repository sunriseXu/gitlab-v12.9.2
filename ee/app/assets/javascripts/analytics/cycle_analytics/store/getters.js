import dateFormat from 'dateformat';
import httpStatus from '~/lib/utils/http_status';
import { dateFormats } from '../../shared/constants';
import { getDurationChartData, getDurationChartMedianData, getTasksByTypeData } from '../utils';

export const hasNoAccessError = state => state.errorCode === httpStatus.FORBIDDEN;

export const currentGroupPath = ({ selectedGroup }) =>
  selectedGroup && selectedGroup.fullPath ? selectedGroup.fullPath : null;

export const selectedProjectIds = ({ selectedProjects }) =>
  selectedProjects.length ? selectedProjects.map(({ id }) => id) : [];

export const cycleAnalyticsRequestParams = ({ startDate = null, endDate = null }, getters) => ({
  project_ids: getters.selectedProjectIds,
  created_after: startDate ? dateFormat(startDate, dateFormats.isoDate) : null,
  created_before: endDate ? dateFormat(endDate, dateFormats.isoDate) : null,
});

export const durationChartPlottableData = state => {
  const { durationData, startDate, endDate } = state;
  const selectedStagesDurationData = durationData.filter(stage => stage.selected);
  const plottableData = getDurationChartData(selectedStagesDurationData, startDate, endDate);

  return plottableData.length ? plottableData : null;
};

export const durationChartMedianData = state => {
  const { durationMedianData, startDate, endDate } = state;
  const selectedStagesDurationMedianData = durationMedianData.filter(stage => stage.selected);
  const plottableData = getDurationChartMedianData(
    selectedStagesDurationMedianData,
    startDate,
    endDate,
  );

  return plottableData.length ? plottableData : [];
};

export const tasksByTypeChartData = ({ tasksByType, startDate, endDate }) => {
  if (tasksByType && tasksByType.data.length) {
    return getTasksByTypeData({
      data: tasksByType.data,
      startDate,
      endDate,
    });
  }
  return { groupBy: [], data: [], seriesNames: [] };
};

const filterStagesByHiddenStatus = (stages = [], isHidden = true) =>
  stages.filter(({ hidden = false }) => hidden === isHidden);

export const hiddenStages = ({ stages }) => filterStagesByHiddenStatus(stages);
export const activeStages = ({ stages }) => filterStagesByHiddenStatus(stages, false);
