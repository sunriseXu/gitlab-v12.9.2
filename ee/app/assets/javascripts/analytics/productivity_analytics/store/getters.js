// eslint-disable-next-line import/prefer-default-export
export const getMetricTypes = state => chartKey =>
  state.metricTypes.filter(m => m.charts.indexOf(chartKey) !== -1);
