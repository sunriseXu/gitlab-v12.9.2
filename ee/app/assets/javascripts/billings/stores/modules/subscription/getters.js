export const isFreePlan = state => state.plan.code === null;

// prevent babel-plugin-rewire from generating an invalid default during karma tests
export default () => {};
