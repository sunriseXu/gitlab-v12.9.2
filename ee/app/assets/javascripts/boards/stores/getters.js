import gettersCE from '~/boards/stores/getters';

export default {
  ...gettersCE,
};
