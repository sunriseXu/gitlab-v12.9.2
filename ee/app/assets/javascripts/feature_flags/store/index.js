import Vue from 'vue';
import Vuex from 'vuex';
import indexModule from './modules/index';
import newModule from './modules/new';
import editModule from './modules/edit';

Vue.use(Vuex);

export const createStore = () =>
  new Vuex.Store({
    modules: {
      index: indexModule,
      new: newModule,
      edit: editModule,
    },
  });

export default createStore();
