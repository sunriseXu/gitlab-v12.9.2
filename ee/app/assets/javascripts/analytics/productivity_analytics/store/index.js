import Vue from 'vue';
import Vuex from 'vuex';
import state from './state';
import * as getters from './getters';
import * as actions from './actions';
import mutations from './mutations';
import filters from './modules/filters/index';
import charts from './modules/charts/index';
import table from './modules/table/index';

Vue.use(Vuex);

const createStore = () =>
  new Vuex.Store({
    state: state(),
    getters,
    actions,
    mutations,
    modules: {
      filters,
      charts,
      table,
    },
  });

export default createStore();
