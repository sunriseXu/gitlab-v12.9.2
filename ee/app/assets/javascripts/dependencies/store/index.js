import Vue from 'vue';
import Vuex from 'vuex';
import listModule from './modules/list';
import * as actions from './actions';
import * as getters from './getters';
import mutations from './mutations';
import state from './state';
import { DEPENDENCY_LIST_TYPES } from './constants';

Vue.use(Vuex);

export default () =>
  new Vuex.Store({
    modules: {
      [DEPENDENCY_LIST_TYPES.all.namespace]: listModule(),
    },
    actions,
    getters,
    mutations,
    state,
  });
