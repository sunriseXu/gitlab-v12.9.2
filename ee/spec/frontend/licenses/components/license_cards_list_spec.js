import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import InstanceCardsList from 'ee/licenses/components/license_cards_list.vue';
import * as getters from 'ee/licenses/store/getters';
import createState from 'ee/licenses/store/state';

describe('InstanceCardsList', () => {
  const newLicensePath = '/newLicensePath';
  let wrapper;

  const localVue = createLocalVue();
  localVue.use(Vuex);

  function createStore(store) {
    const state = Object.assign(createState(), store, {
      newLicensePath,
    });

    return new Vuex.Store({ state, getters });
  }

  function createComponent(store) {
    wrapper = shallowMount(InstanceCardsList, {
      store: createStore(store),
      localVue,
    });
  }

  afterEach(() => {
    if (wrapper) wrapper.destroy();
  });

  it('renders a list of license cards', () => {
    createComponent({ licenses: [{ id: 1 }, { id: 2 }], isLoadingLicenses: false });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders a skeleton loading card if loading licenses', () => {
    createComponent({ isLoadingLicenses: true });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders a message when there are no licenses', () => {
    createComponent({ licenses: [], isLoadingLicenses: false });

    expect(wrapper.element).toMatchSnapshot();
  });
});
