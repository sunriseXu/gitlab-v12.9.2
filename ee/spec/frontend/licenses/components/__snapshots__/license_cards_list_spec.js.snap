// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstanceCardsList renders a list of license cards 1`] = `
<div>
  <div
    class="d-flex justify-content-between align-items-center"
  >
    <h4>
      Instance license
    </h4>
     
    <gl-button-stub
      class="my-3 js-add-license"
      href="/newLicensePath"
      size="md"
      variant="success"
    >
      
      Add license
    
    </gl-button-stub>
  </div>
   
  <ul
    class="license-list list-unstyled"
  >
    <li>
      <license-card-stub
        iscurrentlicense="true"
        license="[object Object]"
      />
    </li>
    <li>
      <license-card-stub
        license="[object Object]"
      />
    </li>
  </ul>
</div>
`;

exports[`InstanceCardsList renders a message when there are no licenses 1`] = `
<div>
  <div
    class="d-flex justify-content-between align-items-center"
  >
    <h4>
      Instance license
    </h4>
     
    <gl-button-stub
      class="my-3 js-add-license"
      href="/newLicensePath"
      size="md"
      variant="success"
    >
      
      Add license
    
    </gl-button-stub>
  </div>
   
  <ul
    class="license-list list-unstyled"
  >
    <li>
      <strong>
        
        No licenses found.
      
      </strong>
    </li>
  </ul>
</div>
`;

exports[`InstanceCardsList renders a skeleton loading card if loading licenses 1`] = `
<div>
  <div
    class="d-flex justify-content-between align-items-center"
  >
    <h4>
      Instance license
    </h4>
     
    <gl-button-stub
      class="my-3 js-add-license"
      href="/newLicensePath"
      size="md"
      variant="success"
    >
      
      Add license
    
    </gl-button-stub>
  </div>
   
  <ul
    class="license-list list-unstyled"
  >
    <li>
      <skeleton-license-card-stub />
    </li>
  </ul>
</div>
`;
